src/prefect/_internal/compatibility/experimental.py

```diff
@@ -179,6 +182,57 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def experimental_field(
+    name: str,
+    *,
+    group: str,
+    help: str = "",
+    stacklevel: int = 2,
+    opt_in: bool = False,
+    when: Optional[Callable[[Any], bool]] = None,
+):
+    """
+    Mark a field in a Pydantic model as experimental.
+
+    Raises warning only if the field is specified during init.
+
+    Example:
+
+        ```python
+
+        @experimental_parameter("y", group="example", when=lambda y: y is not None)
+        def foo(x, y = None):
+            return x + 1 + (y or 0)
+        ```
+    """
+
+    when = when or (lambda _: True)
+
+    @experimental(
+        group=group,
+        feature=f"The field {name!r}",
+        help=help,
+        opt_in=opt_in,
+        stacklevel=stacklevel + 2,
+    )
+    def experimental_check():
+        pass
+
+    def decorator(model_cls: Type[M]) -> Type[M]:
```

Comment:
Not a strong stance, but a short docstring here might be nice. Just to make sure I understand this: we're replacing the models `__init__` method with one that calls the model's `__init__` method and then does a check to see if the attribute specified attribute is present and in the form specified by the `when` function? 

If it's present in the manner specified, it calls the `experimental_check` function, which is decorated the `@experimental` decorator, which gives us the desired warnings and opt-in behavior?

Reply:
Yep, that's right. I added some comments.